### PR TITLE
KOGITO-4091 - Add trusty infinispan and redis images

### PR DESCRIPTION
**Pull request by r00ta**
JIRA: https://issues.redhat.com/browse/KOGITO-4091

This PR aims to add a new kogito image for the trusty service that uses redis as persistence layer.
The previous image `kogito-trusty` is removed, and two new images comes: 
1. `kogito-trusty-infinispan` for the image that uses infinispan
2. `kogito-trusty-redis` for the image that uses redis

Depends on:
kogito-runtimes: https://github.com/kiegroup/kogito-runtimes/pull/949
kogito-apps: https://github.com/kiegroup/kogito-apps/pull/574

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors guide](README.md#contributing-to-kogito-images-repository)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a testcase that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

**Comment by r00ta** [comment]
Depends on https://github.com/kiegroup/kogito-runtimes/pull/949

**Comment by r00ta** [comment]
Hi @radtriste @Kaitou786 , we just merged https://github.com/kiegroup/kogito-apps/pull/574 so this PR aims to add a new trusty image for Redis and Infinispan. 

Once the artifacts are on the nexus, this should be green.. PR for operator is here https://github.com/kiegroup/kogito-cloud-operator/pull/705

File: README.md [hunk 1]
```diff
@@ -591,12 +592,28 @@ to your Kogito infrastructure on a Kubernetes cluster and provide its capabiliti
...
 
+Basic usage with Redis:
+```bash
+$ docker run -it --env KOGITO_PERSISTENCE_REDIS_URL=redis://localhost:6379 quay.io/kiegroup/kogito-trusty-redis:latest
+```
 
 Basic usage
 ```bash
-$ docker run -it -e QUARKUS_INFINISPAN_CLIENT_SERVER_LIST=my-infinispan-server:11222 -e KAFKA_BOOTSTRAP_SERVER=my-kafka:9092 quay.io/kiegroup/kogito-trusty:latest
+$ docker run -it -e QUARKUS_INFINISPAN_CLIENT_SERVER_LIST=my-infinispan-server:11222 -e KAFKA_BOOTSTRAP_SERVER=my-kafka:9092 quay.io/kiegroup/kogito-trusty-infinispan:latest
```
Comment (spolti) [inline README.md]: duplicated basic usage?

File: scripts/run-bats.sh [hunk 2]
```diff
@@ -10,9 +10,6 @@ else
     rm -rf bats-core
 fi 
 
-echo  "----> running bats on kogito-trusty"
```
Comment (spolti) [inline scripts/run-bats.sh:13]: no bats tests for the new images?

File: kogito-trusty-infinispan-overrides.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  value: "Runtime image for Kogito Trusty Service for Infinispan persistence provider"
+- name: "io.k8s.display-name"
+  value: "Kogito Trusty Service - Infinispan"
+- name: "io.openshift.tags"
+  value: "kogito,trusty,trusty-infinspan"
+- name: "io.openshift.expose-services"
+  value: "8080:http"
+
+envs:
+  - name: "SCRIPT_DEBUG"
```
Comment (spolti) [inline kogito-trusty-infinispan-overrides.yaml:17]: maybe its worth to add other envs that the image accepts here?
I know that some images does not have all listed, but it is something to consider.

**Review by spolti — CHANGES_REQUESTED** [review]
minor comments, otherwise looks good

File: scripts/run-bats.sh (hunk 2 above)
Comment (r00ta) [reply scripts/run-bats.sh:13]: Hi @spolti , actually I followed the same structure used for the data index and no bats tests were implemented for such images. If you think we should add them, I'll do it

File: README.md (hunk 1 above)
Comment (r00ta) [reply README.md]: Thanks for the catch!

File: kogito-trusty-infinispan-overrides.yaml (hunk 3 above)
Comment (r00ta) [reply kogito-trusty-infinispan-overrides.yaml:17]: Hi @spolti , I added the env var for the redis image. Regarding the infinspan one: actually the consumer can specify really a lot of env vars (for kafka and infinispan conf), so I'm not sure we should specify all of them. wdyt?

**Review by radtriste — CHANGES_REQUESTED** [review]
I guess you need to remove https://github.com/r00ta/kogito-images/blob/KOGITO-4070-3/kogito-trusty-overrides.yaml ?

**Comment by r00ta** [comment]
> I guess you need to remove https://github.com/r00ta/kogito-images/blob/KOGITO-4070-3/kogito-trusty-overrides.yaml ?

Yep! Thanks for the catch :)

File: scripts/run-bats.sh (hunk 2 above)
Comment (spolti) [reply scripts/run-bats.sh:13]: asking because the old one had a few, isn't needed anymore?

File: kogito-trusty-infinispan-overrides.yaml (hunk 3 above)
Comment (spolti) [reply kogito-trusty-infinispan-overrides.yaml:17]: I think all of those properties are on the docs, right? if so, it is ok to leave as it is.
Or we could add just the basic ones like for, when running it for a very simple scenario locally for tests, wdyt?

@radtriste ^

**Comment by spolti** [comment]
@radtriste does the tests still failing because the missing artifacts on maven?

**Comment by r00ta** [comment]
> @radtriste does the tests still failing because the missing artifacts on maven?

They should be ok now

File: kogito-trusty-infinispan-overrides.yaml (hunk 3 above)
Comment (r00ta) [reply kogito-trusty-infinispan-overrides.yaml:17]: I've added the main env vars for infinispan on the trusty image and the data-index one, let me know if this looks good to you

File: scripts/run-bats.sh (hunk 2 above)
Comment (r00ta) [reply scripts/run-bats.sh:13]: Added the bats tests for the common image that contains the script to enable/disable explainability

**Comment by r00ta** [comment]
Jenkins retest this please

**Comment by r00ta** [comment]
Jenkins retest this please

**Comment by r00ta** [comment]
> @radtriste does the tests still failing because the missing artifacts on maven?

Now I see what you were referring to.. I think CI is failing for reasons not related to these changes

**Comment by r00ta** [comment]
Jenkins retest this please

**Comment by r00ta** [comment]
Jenkins retest this please

**Comment by r00ta** [comment]
Jenkins retest this plase

**Comment by radtriste** [comment]
tests are blocked by https://github.com/kiegroup/kogito-images/pull/378 and/or https://github.com/kiegroup/kogito-runtimes/pull/1115

**Comment by r00ta** [comment]
Hi @radtriste @spolti , according to https://github.com/kiegroup/kogito-images/pull/381 master is broken fyi

**Comment by r00ta** [comment]
Jenkins retest this please

**Review by radtriste — APPROVED** [review]

**Comment by radtriste** [comment]
@spolti good to merge this one ?

**Review by spolti — APPROVED** [review]

**Pull request merged**